specs/appwindow-spec.md

```diff
@@ -0,0 +1,68 @@
+Window.AppWindow api
+===
+New api to simplify accessing [appwindow](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/windowing/windowing-overview) functionality through WinUI code
+
+# Background
+AppWindow provides good set of functionalities for using HWNDs than dealing with win32 apis directly. It is already accessible to winui desktop apps via winappsdk runtime. Currently, users must write a lot of boiler plate code to get to appwindow object and use it. 
+
+[Xaml Window](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window) will expose AppWindow object directly to app developer through an api. Instead of writing a lot of boiler plate code everywhere, app developer can use this api, reducing code bloat, and making appwindow apis easily accessible from winui code.
+
+
+# API Pages
+
+## Window.AppWindow property
+
+This newly introduced api will allow developers to use appwindow object directly within WinUI code.
+```c#
+public Microsoft.UI.Windowing.AppWindow Window.AppWindow { get; }
+```
+
```

Comment:
```suggestion
_Spec note: Mapping an hwnd to an AppWindow causes the hwnd to be subclassed, meaning that the timing of when [AppWindow.GetFromWindowId](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Windowing.AppWindow.GetFromWindowId) is first called can potentially have a behavioral side effect. To ensure this is predictable, the AppWindow for Xaml's hwnd will be created during the Xaml Window's construction. Note that this could potentially be an observable behavior change from the behavior before the introduction of this API._
```

Reply:
Good idea. I have covered some of this effect in the appendix section but an explicit callout is better